app/models/host/managed.rb

```diff
@@ -50,7 +51,7 @@ def initialize(*args)
   end
 
   def build_hooks
-    return unless respond_to?(:old) && old && (build? != old.build?)
+    return if previous_changes['build'].nil?
```

Comment:
this can be nil or an array, empty string / array is not possible for this method.